`/ocm reload` of GApple module fails on 1.13

## Info

- Server version: 1.13.1
- Relevant log excerpt:
```
Caused by: java.lang.IllegalStateException: Duplicate recipe ignored with ID oldcombatmechanics:minecraft
	at net.minecraft.server.v1_13_R2.CraftingManager.a(CraftingManager.java:86) ~[spigot-1.13.1.jar:git-Spigot-0ede7d0-ee5efeb]
	at org.bukkit.craftbukkit.v1_13_R2.inventory.CraftShapedRecipe.addToCraftingManager(CraftShapedRecipe.java:60) ~[spigot-1.13.1.jar:git-Spigot-0ede7d0-ee5efeb]
	at org.bukkit.craftbukkit.v1_13_R2.CraftServer.addRecipe(CraftServer.java:1147) ~[spigot-1.13.1.jar:git-Spigot-0ede7d0-ee5efeb]
	at org.bukkit.Bukkit.addRecipe(Bukkit.java:588) ~[spigot-1.13.1.jar:git-Spigot-0ede7d0-ee5efeb]
	at gvlfm78.plugin.OldCombatMechanics.module.ModuleGoldenApple.registerCrafting(ModuleGoldenApple.java:59) ~[?:?]
	at gvlfm78.plugin.OldCombatMechanics.module.ModuleGoldenApple.reload(ModuleGoldenApple.java:53) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184) ~[?:1.8.0_181]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175) ~[?:1.8.0_181]
```

## Problem Description
Spigot introduced a compatibility layer for materials in 1.13 that rewrites the bytecode to replace Material enum entries with their legacy version. It does however *not* take that into account when finding a recipe, which is why this little safeguard fails:
```java
if(Bukkit.getRecipesFor(napple).size() > 0) return;
Bukkit.addRecipe(r);
```

## Proper solution
Make the target api version 1.13 to disable the rewriting (add `api-version: 1.13` to the plugin.yml).

## Problems with proper solution
* Material.INK_SACK was renamed to Material.INK_SAC, so we'd need to safeguard that. Not sure how to do that scalable, it is late today. I will have a go at that tomorrow unless somebody picks it up.

--- a/pom.xml
+++ b/pom.xml
@@ -32,7 +32,7 @@
         <dependency>
             <groupId>org.bukkit</groupId>
             <artifactId>bukkit</artifactId>
-            <version>1.12.2-R0.1-SNAPSHOT</version>
+            <version>1.13.1-R0.1-SNAPSHOT</version>
         </dependency>
         <dependency>
             <groupId>net.gravitydevelopment.updater</groupId>

--- a/src/main/java/com/codingforcookies/armourequip/ArmourType.java
+++ b/src/main/java/com/codingforcookies/armourequip/ArmourType.java
@@ -3,10 +3,7 @@ package com.codingforcookies.armourequip;
 import org.bukkit.inventory.ItemStack;
 
 /**
- * @Author Borlea
- * @Github https://github.com/borlea/
- * @Website http://codingforcookies.com/
- * @Since Jul 30, 2015 6:46:16 PM
+ * Enum declarations by (<a href="http://codingforcookies.com/">Barlea</a>), methods by us.
  */
 public enum ArmourType {
     HELMET(5), CHESTPLATE(6), LEGGINGS(7), BOOTS(8);
@@ -27,33 +24,19 @@ public enum ArmourType {
         if(itemStack == null){
             return null;
         }
-        switch(itemStack.getType()){
-            case DIAMOND_HELMET:
-            case GOLD_HELMET:
-            case IRON_HELMET:
-            case CHAINMAIL_HELMET:
-            case LEATHER_HELMET:
-                return HELMET;
-            case DIAMOND_CHESTPLATE:
-            case GOLD_CHESTPLATE:
-            case IRON_CHESTPLATE:
-            case CHAINMAIL_CHESTPLATE:
-            case LEATHER_CHESTPLATE:
-                return CHESTPLATE;
-            case DIAMOND_LEGGINGS:
-            case GOLD_LEGGINGS:
-            case IRON_LEGGINGS:
-            case CHAINMAIL_LEGGINGS:
-            case LEATHER_LEGGINGS:
-                return LEGGINGS;
-            case DIAMOND_BOOTS:
-            case GOLD_BOOTS:
-            case IRON_BOOTS:
-            case CHAINMAIL_BOOTS:
-            case LEATHER_BOOTS:
-                return BOOTS;
-            default:
-                return null;
+
+        String typeName = itemStack.getType().name();
+
+        if(typeName.endsWith("_HELMET")){
+            return HELMET;
+        } else if(typeName.endsWith("_CHESTPLATE")){
+            return CHESTPLATE;
+        } else if(typeName.endsWith("_LEGGINGS")){
+            return LEGGINGS;
+        } else if(typeName.endsWith("_BOOTS")){
+            return BOOTS;
+        } else {
+            return null;
         }
     }
 

--- a/src/main/java/gvlfm78/plugin/OldCombatMechanics/module/ModuleNoLapisEnchantments.java
+++ b/src/main/java/gvlfm78/plugin/OldCombatMechanics/module/ModuleNoLapisEnchantments.java
@@ -15,9 +15,16 @@ import org.bukkit.permissions.Permissible;
 
 public class ModuleNoLapisEnchantments extends Module {
 
+    private Material lapisLazuliMaterial;
 
     public ModuleNoLapisEnchantments(OCMMain plugin){
         super(plugin, "no-lapis-enchantments");
+
+        try{
+            lapisLazuliMaterial = Material.valueOf("INK_SACK");
+        } catch(IllegalArgumentException ignored){
+            lapisLazuliMaterial = Material.LAPIS_LAZULI;
+        }
     }
 
     @EventHandler
@@ -40,10 +47,16 @@ public class ModuleNoLapisEnchantments extends Module {
         if(!hasPermission(e.getWhoClicked())) return;
 
         ItemStack item = e.getCurrentItem();
-        if(item != null && (
-                (item.getType() == Material.INK_SACK && e.getRawSlot() == 1) ||
-                        (e.getCursor() != null && e.getCursor().getType() == Material.INK_SACK && e.getClick() == ClickType.DOUBLE_CLICK)))
+
+        if(item == null){
+            return;
+        }
+
+        if(item.getType() == lapisLazuliMaterial && e.getRawSlot() == 1){
             e.setCancelled(true);
+        } else if(e.getCursor() != null && e.getCursor().getType() == lapisLazuliMaterial && e.getClick() == ClickType.DOUBLE_CLICK){
+            e.setCancelled(true);
+        }
     }
 
     @EventHandler

--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -6,6 +6,7 @@ description: Reverts to pre-1.9 combat mechanics
 website: http://dev.bukkit.org/bukkit-plugins/oldcombatmechanics
 load: POSTWORLD
 softdepend: [PlaceholderAPI, Spartan]
+api-version: 1.13
 
 commands:
   OldCombatMechanics:

